File: BinarytoDecimal.r
BinaryToDecimal <- function(binaryNumber) {
	value = toString(binaryNumber)
	length = nchar(value)
	output = 0
	for (position in 1:length) {
		char = substr(value, position, position)
		if (char == "1") {
			output = output + (2 ^ (length - position))
		} else if (char == "0") {
			
		} else {
			cat("Make sure you only input binary values", "\n")
			return(-1)
		}
	}
	return(output)
}

cat("Binary Number: ")
num <- readLines("stdin",n=1)
iserror = BinaryToDecimal(num)
if (iserror != -1) {
	cat(iserror, "\n")
}

File: BinarytoDecimal.f90
! gfortran BinarytoDecimal.f90 -o BinarytoDecimal

program BinaryDecimal
    implicit none
    integer num
    write(*,'(a,$)') "Binary Number: "
    read(*,*) num
    write(*, "(I0)") binary_to_decimal(num)
contains

integer function binary_to_decimal(binaryNumber)
    integer, intent(in) :: binaryNumber
    character(len = 8) :: value
    integer :: length = len(value)
    integer :: output = 0
    character(len = 1) :: char
    integer :: position = 0
    write(value, "(I8)") binaryNumber
    do position = 1, length
        char = value(position:position+1)
        if (char == '1') then
            output = output + (2.0**(length - position))
        else if (char == '0' .or. char == " ") then
            cycle
        else
            stop 'Make sure you only input binary values'
        end if
    end do
    binary_to_decimal = output
end function

end program BinaryDecimal
